File: Java/Leetcode/SlidingWindow/MinSizeSubSum.java
package com.brianprojects.Leetcode.SlidingWindow;

public class MinSizeSubSum {
    //TIme Complexity: O(n) || Space Complexity: O(1)
    public int minSubArrayLen(int target, int[] nums){
        if(target == 0 || nums == null){
            return 0;
        }

        int min = Integer.MAX_VALUE;
        int total = nums[0];
        int r = 0;

        for(var i = 0; i<nums.length; i++){
            if(min == 1){
                return 1;
            }
            while(r!=nums.length-1 && total<target){
                total+=nums[r+1];
                r++;
            }
            if(total >= target){
                min = Math.min(min, (r-i)+1);
                total-=nums[i];
            }
        }
        return min == Integer.MAX_VALUE? 0: min;
    }
}

File: Python/Leetcode_Python/SlidingWindow/MinSizeSubSum.py
from typing import List


def minSubArrayLen(target: int, nums: List[int]) -> int:
    if not nums:
        return -1

    l = 0
    r = 0

    total = 0

    minimum = 1000000009

    while r < len(nums):
        val = nums[r]
        if total < target:
            total += val
            r += 1
        else:
            minimum = min(minimum, (r - l) + 1)
            total -= nums[l]
            l += 1
            if l > r:
                r = l

    return minimum
